File: src/main/java/net/botlify/anticaptchacom/CaptchaType.java
package net.botlify.anticaptchacom;

/**
 * The type of the captcha to solve.
 */
public enum CaptchaType {

    /**
     * The Google reCAPTCHA v2 type.
     */
    RECAPTCHA_V2,

    /**
     * @deprecated This type is not supported.
     */
    @Deprecated
    RECAPTCHA_V3,

    /**
     * @deprecated This type is not supported.
     */
    @Deprecated
    IMAG_CAPTCHA,

    /**
     * @deprecated This type is not supported.
     */
    @Deprecated
    RECAPTCHA_ENTERPRISE_V2_OR_V3,

    /**
     * @deprecated This type is not supported.
     */
    @Deprecated
    HCAPTCHA,

    /**
     * @deprecated This type is not supported.
     */
    @Deprecated
    GEETEST,

    /**
     * @deprecated This type is not supported.
     */
    @Deprecated
    ARKOSE_LABS,

    /**
     * @deprecated This type is not supported.
     */
    @Deprecated
    CUSTOM_TASKS,

    /**
     * @deprecated This type is not supported.
     */
    @Deprecated
    TURNSTILE;

}

File: src/main/kotlin/net/botlify/anticaptchacom/enums/CaptchaType.kt
package net.botlify.anticaptchacom.enums

/**
 * The type of the captcha to solve.
 */
enum class CaptchaType {
    /**
     * The Google reCAPTCHA v2 type.
     */
    RECAPTCHA_V2,


    @Deprecated("This type is not supported.")
    RECAPTCHA_V3,


    @Deprecated("This type is not supported.")
    IMAG_CAPTCHA,


    @Deprecated("This type is not supported.")
    RECAPTCHA_ENTERPRISE_V2_OR_V3,


    @Deprecated("This type is not supported.")
    HCAPTCHA,


    @Deprecated("This type is not supported.")
    GEETEST,


    @Deprecated("This type is not supported.")
    ARKOSE_LABS,


    @Deprecated("This type is not supported.")
    CUSTOM_TASKS,


    @Deprecated("This type is not supported.")
    TURNSTILE
}
